stm/stm.ml

```diff
@@ -2097,12 +2097,21 @@ let known_state ~doc ?(redefine_qed=false) ~cache id =
                         | None ->
                           CErrors.anomaly Pp.(str "Attempting to force admitted proof contents."))
                     in
-                    let proof =
-                      PG_compat.close_future_proof ~feedback_id:id fp' in
+                    let control, pe = extract_pe x in
+                    let control = VernacControl.from_syntax control in
+                    let control, proof =
+                      VernacControl.under_control ~loc:x.expr.loc
+                        ~with_local_state:VernacControl.trivial_state
+                        control
+                        ~noop:None
+                        (fun () -> Some (PG_compat.close_future_proof ~feedback_id:id fp'))
+                    in
+                    (* We only get [noop] from [Fail],
+                       but we can't see [Fail] in this classification. *)
```

Comment:
I don't get this message. Is noop actually None?

Reply:
I think I see now, but then I believe it could happen.

Reply:
How could it happen?

Reply:
Maybe I'm mistaken, but this is what I would try.
```
Lemma x : True.
Proof.
trivial.
Fail Qed.
```
If it works, just ignore my remark.

Reply:
It works
It also works with no `trivial` (which is the noop case)